Sources/HDWallet/DerivationPath.swift

```diff
@@ -7,15 +7,78 @@
 import Foundation
 
 /// Represents a hierarchical determinisic derivation path.
-public struct DerivationPath: Hashable, CustomStringConvertible {
+public struct DerivationPath: Codable, Hashable, CustomStringConvertible {
+    let indexCount = 5
+
     /// List of indices in the derivation path.
     public private(set) var indices = [Index]()
 
+    /// Address purpose, each blockchain will have a different value.
+    public var purpose: Int? {
+        get {
+            return indices[0].value
+        }
+        set {
+            indices[0] = Index(newValue ?? 0, hardened: true)
+        }
+    }
+
+    /// Coin type distinguishes between main net, test net, and forks.
+    public var coinType: Int? {
+        get {
+            return indices[1].value
+        }
+        set {
+            indices[1] = Index(newValue ?? 0, hardened: true)
+        }
+    }
+
+    /// Account number.
+    public var account: Int? {
+        get {
+            return indices[2].value
+        }
+        set {
+            indices[2] = Index(newValue ?? 0, hardened: true)
+        }
+    }
+
+    /// Change or private addresses will set this to 1.
+    public var change: Int? {
+        get {
+            return indices[3].value
+        }
+        set {
+            indices[3] = Index(newValue ?? 0, hardened: false)
+        }
+    }
+
+    /// Address number
+    public var address: Int? {
+        get {
+            return indices[4].value
+        }
+        set {
+            indices[4] = Index(newValue ?? 0, hardened: false)
+        }
+    }
+
     init(indices: [Index]) {
+        precondition(indices.count == indexCount, "Not enough indices")
         self.indices = indices
     }
 
-    /// Initializes a derivation path with a string description like `m/10/0/2'/3`
+    /// Creates a `DerivationPath` by components.
+    public init(purpose: Int, coinType: Int, account: Int, change: Int, address: Int) {
```

Comment:
`account ` and `change` not used much in our code, maybe set default value to `0`?

Reply:
ok